base_classes/nyaml/NXopt_assembly.yaml

```diff
@@ -0,0 +1,46 @@
+category: base 
+doc: |
+  Contains datastructures of an experimental optical setup (i.e Multiple
+  Transfermatrix Tables).
+  These datastructure, are used to relate physical properties of two beams
+  (NXopt_beam) which have one common optical element (NXopt_element) (one 
+  specific transfermatrix).
+  One of these beams in an input beam and the other one is an output beam.
+
+  The data describes the change of beam properties, i.e. The intensity of a
+  beam is reduced, because the transmission coefficient of the opt element is 
+  lower than 1.
+  
+opt_assembly(NXdata):
+  datatype_N:
+    doc: |
+      Select which type of data was recorded, for example Psi and Delta
+      (see: https://en.wikipedia.org/wiki/Ellipsometry#Data_acquisition).
+      It is possible to have multiple selections. Data types may also be
+      converted to each other, e.g. a Mueller matrix contains N,C,S data
+      as well. This selection defines how many columns (N_variables) are
+      stored in the data array.
+      N in the name, is the index number in which order the given
+      property is listed. 
+    enumeration:

```

Comment:
The examples `Psi` and `Delta` are not possible with the enumeration.

In general, all your fields will be called `datatype_<your_name>, so you cannot use anything that doesn`t start with `datatype_`.

Reply:
Just replaced the respective examples - and shortened the other text, as I don't understand the given Mueller matrix example.